pennylane/debugging.py

```diff
@@ -105,3 +108,75 @@ def get_snapshots(*args, **kwargs):
         return dbg.snapshots
 
     return get_snapshots
+
+
+class PLDB(pdb.Pdb):
+    """Custom debugging class integrated with Pdb."""
+
+    __active_dev = []
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the debugger, and set custom prompt string."""
+        super().__init__(*args, **kwargs)
+        self.prompt = "[pldb]: "
+
+    @classmethod
+    def valid_context(cls):
+        """Determine if the debugger is called in a valid context.
+
+        Raises:
+            TypeError: Can't call breakpoint outside of a qnode execution
+            TypeError: Device not supported with breakpoint
+        """
+
+        if not qml.queuing.QueuingManager.recording() or not cls.is_active_dev():
+            raise TypeError("Can't call breakpoint outside of a qnode execution")
+
+        if cls.get_active_device().name not in ("default.qubit", "lightning.qubit"):
```

Comment:
# Question
Is this because there is something specific about the _implementation_ of these devices? Or just because they have a certain interface? If it's the latter, we should define the interface using a [`Protocol`](https://peps.python.org/pep-0544/) decorated with [`runtime_checkable()`](https://docs.python.org/3/library/typing.html#typing.runtime_checkable) and simply check if the device supports the interface here.

Reply:
It's definitely due to the interface. However, we want to restrict this feature to these two devices for now. For now I will leave the implementation as is, once we decide to open it to all devices with this interface we can revisit this suggestion!

Thanks,